forte/data/ontology/top.py

```diff
@@ -883,26 +883,41 @@ def image(self):
 
     @property
     def max_x(self):
-        return self._image_width - 1
+        return self.image_shape[1] - 1
 
     @property
     def max_y(self):
-        return self._image_height - 1
+        return self.image_shape[0] - 1
 
-    def set_image_shape(self, width, height):
+    @property
+    def image_shape(self):
         """
-        This function is used to set the shape of the image.
+        Returns the shape of the image.
 
-        Args:
-            width: the width of the image. The unit is pixel.
-            height: the height of the image. The unit is pixel.
+        Raises:
+            ValueError: if the image annotation is not attached to any data
+                pack.
+            ValueError: if the image shape is not valid. It must be either
+                2D or 3D.
+
+        Returns:
+            The shape of the image.
         """
-        self._image_width = (  # pylint: disable=attribute-defined-outside-init
-            width
-        )
-        self._image_height = (  # pylint: disable=attribute-defined-outside-init
-            height
-        )
+        if self.pack is None:
+            raise ValueError(
+                "Cannot get image because image annotation is not "
+                "attached to any data pack."
+            )
+        image_shape = self.pack.get_payload_at(
+            Modality.Image, self._image_payload_idx
+        )._cache_shape
```

Comment:
Maybe it's better to use `image_shape` here.

Reply:
do you mean change the variable name? `cache_shape` -> `image_shape`. `cache_shape` is more generic variable for payloads of all modalities.

Reply:
No you don't need to change the variable name. I think `Payload` already exposed a property function called `image_shape` right?